Digging_to_kernel

wumingzhilian opened the pull request:
Digging_to_kernel

wumingzhilian committed:
Digging_to_kernel

Digging_to_kernel

mudongliang commented on _posts/2022-06-07-RealWorldCTF-digging_into_kernel.md:
```diff
@@ -0,0 +1,1114 @@
+# 2022-RealWorld-CTF(experience competition)-digger_into_kernel
```
Refer to [CISCN2017 - babydriver](https://github.com/mudongliang/mudongliang.github.io/edit/master/_posts/2022-01-18-ciscn2017---babydriver.md) and modify the titles and subtitles, e.g., "# -> ##, ##->###". And then add the first ten lines of the referred post here.

wumingzhilian replied:
Thank you very much for your reply, the article has been revised.

wumingzhilian replied:
References：
[1] https://mp.weixin.qq.com/s/v9_C43Qh9qXr9tYojrZKgQ
[2] https://xz.aliyun.com/t/11053
[3] https://mudongliang.github.io/2022/01/18/ciscn2017-babydriver.html
[4]https://github.com/g0dA/linuxStack/blob/master/%E5%85%B3%E4%BA%8E%E5%86%85%E5%AD%98%E5%88%86%E9%85%8D%E7%9A%84%E4%B8%80%E4%BA%9B%E4%BA%8B.md
[5] 段鑫峰. 面向Linux内核空间的内存分配隔离方法的研究与实现[D].北京交通大学,2021.DOI:10.26944/d.cnki.gbfju.2021.000664.
[6] https://www.anquanke.com/post/id/259252#h2-6
[7] https://zhuanlan.zhihu.com/p/490588193
[8] https://kernel.blog.csdn.net/article/details/52705552
[9]https://www.jeanleo.com/2018/09/07/%E3%80%90linux%E5%86%85%E5%AD%98%E6%BA%90%E7%A0%81%E5%88%86%E6%9E%90%E3%80%91slub%E5%88%86%E9%85%8D%E7%AE%97%E6%B3%95%EF%BC%883%EF%BC%89/

wumingzhilian committed:
Update 2022-06-07-RealWorldCTF-digging_into_kernel.md

References:
[1] https://mp.weixin.qq.com/s/v9_C43Qh9qXr9tYojrZKgQ
[2] https://xz.aliyun.com/t/11053
[3] https://mudongliang.github.io/2022/01/18/ciscn2017-babydriver.html
[4]https://github.com/g0dA/linuxStack/blob/master/%E5%85%B3%E4%BA%8E%E5%86%85%E5%AD%98%E5%88%86%E9%85%8D%E7%9A%84%E4%B8%80%E4%BA%9B%E4%BA%8B.md
[5]段鑫峰. 面向Linux内核空间的内存分配隔离方法的研究与实现[D].北京交通大学,2021.DOI:10.26944/d.cnki.gbfju.2021.000664.
[6] https://www.anquanke.com/post/id/259252#h2-6
[7] https://zhuanlan.zhihu.com/p/490588193
[8] https://kernel.blog.csdn.net/article/details/52705552
[9]https://www.jeanleo.com/2018/09/07/%E3%80%90linux%E5%86%85%E5%AD%98%E6%BA%90%E7%A0%81%E5%88%86%E6%9E%90%E3%80%91slub%E5%88%86%E9%85%8D%E7%AE%97%E6%B3%95%EF%BC%883%EF%BC%89/

mudongliang commented on _posts/2022-06-07-RealWorldCTF-digging_into_kernel.md:
```diff
@@ -0,0 +1,1128 @@
+
+To facilitate debugging, we can use the gdb parameter, `-gdb tcp::1234` ,to open the debugging port.
+
+This `run.sh ` scripts don't contain the argument of `-monitor null` , so you can enter the monitor mode by pressing `ctrl+A` and then entering C (Ps: the command execution of the host can be equivalent to escape)
```
This `run.sh` script does not ...

wumingzhilian replied:
Thank you for your correction!

mudongliang commented on _posts/2022-06-07-RealWorldCTF-digging_into_kernel.md:
```diff
@@ -0,0 +1,1128 @@
+
+#### Trick
+
+https://elixir.bootlin.com/linux/v5.4.38/source/include/linux/sched.h#L878
```
<https://elixir.bootlin.com/linux/v5.4.38/source/include/linux/sched.h#L878>

mudongliang commented on _posts/2022-06-07-RealWorldCTF-digging_into_kernel.md:
```diff
@@ -0,0 +1,1128 @@
+
+This structure is used to apply and generate the cred structure.
+
+https://elixir.bootlin.com/linux/v5.4.38/source/kernel/cred.c#L250
```
<https://elixir.bootlin.com/linux/v5.4.38/source/kernel/cred.c#L250>

And the same operation to the single url.

mudongliang commented on _posts/2022-06-07-RealWorldCTF-digging_into_kernel.md:
````diff
@@ -0,0 +1,1128 @@
+	……
+```
+
+Analysis of `prepare_creds`:
````
You need to add analysis of the function [cred_init](https://elixir.bootlin.com/linux/latest/source/kernel/cred.c#L689). This helps you verify the length of this dedicated slab.

wumingzhilian replied:
Thank you very much for your correction, the analysis of the cred_init function has been added.

wumingzhilian replied:
Thank you very much for your corrections, which have added analysis of cred_init functions.

wumingzhilian committed:
Update 2022-06-07-RealWorldCTF-digging_into_kernel.md

wumingzhilian commented on _posts/2022-06-07-RealWorldCTF-digging_into_kernel.md:
```diff
@@ -0,0 +1,1147 @@
+
+The cred structure contains the `permission information` of each process,such as `uid`, `gid`,`suid`, etc.
+
+https://elixir.bootlin.com/linux/v5.4.38/source/include/linux/cred.h#L111
```
https://elixir.bootlin.com/linux/v5.4.38/source/include/linux/cred.h#L111

wumingzhilian commented on _posts/2022-06-07-RealWorldCTF-digging_into_kernel.md:
```diff
@@ -0,0 +1,1147 @@
+
+Analysis of `cred_init` :
+
+https://elixir.bootlin.com/linux/v5.4.38/source/kernel/cred.c#L656
```
https://elixir.bootlin.com/linux/v5.4.38/source/kernel/cred.c#L656

wumingzhilian committed:
Update 2022-06-07-RealWorldCTF-digging_into_kernel.md
